x-pack/plugins/security_solution/server/endpoint/services/artifacts/manifest_manager/manifest_manager.ts

```diff
@@ -835,4 +867,241 @@ export class ManifestManager {
       );
     }
   }
+
+  /**
+   * Unified Manifest methods
+   */
+
+  private setNewSemanticVersion(semanticVersion: string): string | null {
+    const newSemanticVersion = semver.inc(semanticVersion, 'patch');
+    if (!semver.valid(newSemanticVersion)) {
+      throw new Error(`Invalid semver: ${newSemanticVersion}`);
+    }
+    return newSemanticVersion;
+  }
+
+  protected getUnifiedManifestClient(): UnifiedManifestClient {
+    return new UnifiedManifestClient(this.savedObjectsClient);
+  }
+
+  public async getAllUnifiedManifestsSOFromCache(): Promise<InternalUnifiedManifestSchema[]> {
```

Comment:
I made them public so they can be easily tested. I chose this approach because all these methods involve a lot of logic, and keeping them separate makes them easier to understand. However, I still want to be able to test them independently from the public methods that use them. Ultimately, I decided to make them public because this class is only used internally and isn't accessed by any API. I don't think they need to be strictly "protected."